WooCommerce/Classes/Tools/UnitInputFormatter/PriceInputFormatter.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+
+/// `UnitInputFormatter` implementation for decimal number input.
+///
+struct PriceInputFormatter: UnitInputFormatter {
+
+    /// Currency Formatter.
+    ///
+    private var currencyFormatter = CurrencyFormatter()
+
+    /// Number formatter with comma
+    ///
+    private let numberFormatterPoint: NumberFormatter = {
+        let numberFormatter = NumberFormatter()
+        numberFormatter.decimalSeparator = "."
+        numberFormatter.numberStyle = .decimal
+        return numberFormatter
+    }()
+
+    /// Number formatter with comma
+    ///
+    private let numberFormatterComma: NumberFormatter = {
+        let numberFormatter = NumberFormatter()
+        numberFormatter.decimalSeparator = ","
+        numberFormatter.numberStyle = .decimal
+        return numberFormatter
+    }()
+
+    func isValid(input: String) -> Bool {
+        guard input.isEmpty == false else {
+            // Allows empty input to be replaced by 0.
+            return true
+        }
+
+        return numberFormatterPoint.number(from: input) != nil || numberFormatterComma.number(from: input) != nil
+    }
+
+    func format(input text: String?) -> String {
+        guard let text = text, text.isEmpty == false else {
+            return "0"
+        }
+
+        // Replace point or comma with the current decimal separator configured on website
+        var formattedText = text.replacingOccurrences(of: ".", with: CurrencySettings.shared.decimalSeparator)
+        formattedText = formattedText.replacingOccurrences(of: ",", with: CurrencySettings.shared.decimalSeparator)
```

Comment:
I've been thinking about how we can format the price/dimension/weight values with the same validation on web, and saw the configuration of [`decimalSeparator`](https://developer.apple.com/documentation/foundation/numberformatter/1408029-decimalseparator) and [`hasThousandSeparators`](https://developer.apple.com/documentation/foundation/numberformatter/1416451-hasthousandseparators) in `NumberFormatter` - what do you think of setting `NumberFormatter`'s `decimalSeparator` with `CurrencySettings.shared.decimalSeparator` and `hasThousandSeparators` to false (since web does not allow thousand separators), then using `NumberFormatter` to format the input text here?

Reply:
regarding the decimal separator from the decimal pad keyboard, I'm planning on replacing any characters not in the set of 0-9 and the site's `decimalSeparator` (like the comma case) with `decimalSeparator` - what do you think?

Reply:
Unfortunately, `hasThousandSeparators` in NumberFormatter is available only in the macOS SDK (why 🤷‍♂ ).

> regarding the decimal separator from the decimal pad keyboard, I'm planning on replacing any characters not in the set of 0-9 and the site's decimalSeparator (like the comma case) with decimalSeparator - what do you think?

This can make sense, and it can work well also with the copy/paste. I updated the code accordingly. 👍